vme/src/skills.cpp

```diff
@@ -1051,49 +1053,37 @@ bool pairISCompare(const std::pair<int, std::string> &firstElem, const std::pair
 
 void ability_dump(void)
 {
-    std::string str;
-    char buf[MAX_STRING_LENGTH];
-
-    for (int j = 0; j < PROFESSION_MAX; j++)
+    for (int profession_index = 0; profession_index < PROFESSION_MAX; profession_index++)
     {
         std::vector<std::pair<int, std::string>> vect;
 
         for (int i = 0; i < ABIL_TREE_MAX; i++)
         {
-            str = "";
-
-            snprintf(buf, sizeof(buf), "%s,%s", g_AbiColl.text[i], spc(20 - strlen(g_AbiColl.text[i])));
-            str.append(buf);
+            auto str = diku::format_to_str("%s,%s", g_AbiColl.text[i], spc(20 - strlen(g_AbiColl.text[i])));
 
-            snprintf(buf,
-                     sizeof(buf),
-                     ".profession %s%s = %s%d\n",
-                     g_professions[j],
-                     spc(12 - strlen(g_professions[j])),
-                     (g_AbiColl.prof_table[i].profession_cost[j] >= 0) ? "+" : "",
-                     g_AbiColl.prof_table[i].profession_cost[j]);
-            str.append(buf);
+            str += diku::format_to_str(".profession %s%s = %s%d\n",
+                                       g_professions[profession_index],
+                                       spc(12 - strlen(g_professions[profession_index])),
+                                       (g_AbiColl.prof_table[i].profession_cost[profession_index] >= 0) ? "+" : "",
+                                       g_AbiColl.prof_table[i].profession_cost[profession_index]);
 
-            vect.push_back(std::make_pair(g_AbiColl.prof_table[i].profession_cost[j], str));
+            vect.push_back(std::make_pair(g_AbiColl.prof_table[i].profession_cost[profession_index], str));
```

Comment:
Should the vector push_back be moved down to 1088, otherwise stuff is added to `str` and never used as it is wiped on the next loop iteration.